### PR TITLE
Remove unused information in API_OVERVIEW file

### PR DESCRIPTION
**Pull request by hnwolf**
Information and link in API Overview file is not availabe now so I remove it.

### CODE REVIEW

File: api/API_OVERVIEW.md [hunk 1]
```diff
@@ -76,8 +76,7 @@ See [here](CONTRIBUTING.md#api-changes) for the v2 API change process.
 Unless otherwise stated, the APIs with the same names as v1 APIs have a similar role.
```
Comment (alyssawilk) [inline api/API_OVERVIEW.md]: we can remove this sentence, since the v1 APIs are deprecated

File: api/API_OVERVIEW.md [hunk 2]
```diff
@@ -76,8 +76,7 @@ See [here](CONTRIBUTING.md#api-changes) for the v2 API change process.
...
 
 * [Cluster Discovery Service (CDS)](envoy/api/v2/cds.proto).
```
Comment (alyssawilk) [inline api/API_OVERVIEW.md]: This API supports dynamically fetching cluster manager members.  See the CDS documentation

with a link to https://www.envoyproxy.io/docs/envoy/latest/configuration/cluster_manager/cds.html

File: api/API_OVERVIEW.md [hunk 3]
```diff
@@ -76,8 +76,7 @@ See [here](CONTRIBUTING.md#api-changes) for the v2 API change process.
...
-* [Endpoint Discovery Service (EDS)](envoy/api/v2/eds.proto). This has the same role as SDS in the [v1 API](https://www.envoyproxy.io/docs/envoy/latest/api-v1/cluster_manager/sds),
-  the new name better describes what the API does in practice. Advanced global load balancing capable of utilizing N-dimensional upstream metrics is now supported.
+* [Endpoint Discovery Service (EDS)](envoy/api/v2/eds.proto). The new name better describes what the API does in practice. Advanced global load balancing capable of utilizing N-dimensional upstream metrics is now supported.
```
Comment (alyssawilk) [inline api/API_OVERVIEW.md]: This API supports dynamically fetching endpoints. See the EDS documentation

with a link to https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/service_discovery#arch-overview-service-discovery-types-eds

File: api/API_OVERVIEW.md [hunk 4]
```diff
@@ -76,8 +76,7 @@ See [here](CONTRIBUTING.md#api-changes) for the v2 API change process.
...
 * [Health Discovery Service (HDS)](envoy/service/discovery/v2/hds.proto). This new API supports efficient endpoint health discovery by the management server via the Envoy instances it manages. Individual Envoy instances
```
Comment (alyssawilk) [inline api/API_OVERVIEW.md]: And please remove "new" from "This new API" here and below

**Comment by mattklein123** [comment]
Closing out. Please reopen a single PR with all of these fixes for review.

**Pull request closed without merging**